yjit/src/asm/x86_64/mod.rs

```diff
@@ -606,7 +606,7 @@ fn write_rm_multi(cb: &mut CodeBlock, op_mem_reg8: u8, op_mem_reg_pref: u8, op_r
         },
         // R/M + UImm
         (_, X86Opnd::UImm(uimm)) => {
-            let num_bits = imm_num_bits(uimm.value.try_into().unwrap());
+            let num_bits = uimm_num_bits(uimm.value.try_into().unwrap());
```

Comment:
@XrXr I'm assuming we're using `imm_num_bits` because of sign-extension here as well?

We should probably add a comment to explain that 🤔 , it is pretty unintuitive.

Reply:
Yeah, it's probably due to accounting for sign extension. Though in this case I think there is no right answer because in this place we don't know whether the instruction we're emitting does sign extension or not. The caller should be able to decide this 🤔

Reply:
Actually, we do have operand size information here, and it looks like when it comes to immediates all of them are sign extended. So the correct logic should be something like:
```
if lhs.size > uimm.bit_size {
  bit_size = imm_num_bits(uimm); // the bit sizes are different, so it's going to get sign extended
} else {
  bit_size = uimm.bit_size; // imm size same as the size of the other operand, no extension happens.
}
```